AndroidSDKCore/src/main/java/com/leanplum/callbacks/MessageDisplayedCallback.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2013, Leanplum, Inc. All rights reserved.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.leanplum.callbacks;
+
+import com.leanplum.ActionContext;
+
+/**
+ * Message displayed callback.
+ *
+ * @author Mayank Sanganeria
+ */
+public abstract class MessageDisplayedCallback implements Runnable {
+
+  private ActionContext actionContext;
+
+  public void setActionContext(ActionContext actionContext) {
+    this.actionContext = actionContext;
+  }
+
+  public void run() {
+    this.messageDisplayed(actionContext);
+  }
+
+  public abstract void messageDisplayed(ActionContext actionContext);
```

Comment:
Does `ActionContext` mean anything to the developer consuming this function? Since this is specifically for archiving messages, maybe we want something that makes more sense, like a `MessageContext` with members for each business requirement of the archival feature?

Reply:
That's something I was thinking about too. 
Either 
1. this callback ends up being specifically for archiving messages OR
2. this is a more general callback with more information than just archiving messages

Since it's a customer facing callback, if more information is needed later, making changes and deprecating becomes a bit messy.

I was thinking we enable the most info possible to be accessible through the callback and  document the things specifically needed for the archiving use case. That way if we have other use cases or if someone needs more information for archiving, it all continues to funnel through the same callback.

Thoughts?

Reply:
I like the idea of having all information given in a generic, more scalable way. However, in reality we have only one exact use-case. We know exactly what is needed for a specific purpose. I don't want to have a paragraph of documentation explaining how to use a generic API that was only created for this specific use-case. Less setup, more plug-and-play, more delighting our customer.

At the risk of scope-creep... Could we add a function to ActionContext like `getBlob/Archive/FINRA/etc...` that returns what is needed for this exact use-case?

Otherwise I would lean toward option 1. If we ever need something more generic we won't have to deprecate 1. We can just add the more generic one.